### ISSUE TITLE
Invalid class_weight keys silently ignored in model.fit(), no error raised

### ISSUE THREAD

**Issue by MrNagabhushana-0-dev** [keras-team/keras#23220]
When passing an invalid `class_weight` dictionary (with a key that doesn't match any class index) into `model.fit()`, no error or warning is raised. Training proceeds silently as if `class_weight` was not provided, which can hide real bugs in user code.

**Reproduction:**

import tensorflow as tf
from tensorflow.keras import layers

class MyModel(tf.keras.Model):
    def __init__(self):
        super().__init__()
        self.dense = layers.Dense(4, activation='relu')

    def call(self, inputs):
        return self.dense(inputs)

model = MyModel()
model.compile(optimizer='adam', loss='mse')

x = tf.random.normal((10, 1))
y = tf.ones((10, 1))

model.fit(x, y, class_weight={'invalid': 0.5}, verbose=0)
# No error raised — expected ValueError or KeyError

**Expected behavior:**
Invalid class_weight keys should raise a `ValueError` or `KeyError`.

**Context:**
Originally reported at tensorflow/tensorflow#98283. A fix was attempted at tensorflow/tensorflow#98429, but the maintainer closed it noting that class_weight validation logic should live in the Keras repo instead of the legacy tf.keras path in TensorFlow core.

**Comment by maitry63** [comment]
Hi @MrNagabhushana-0-dev , thanks for reporting this issue!

Reproduction Bug Notebook - [gist](https://colab.sandbox.google.com/gist/maitry63/dbfb78749c6d1508383a202bee01ea93/untitled7.ipynb#scrollTo=HpaRBFJcFK_4)

**Comment by MrNagabhushana-0-dev** [comment]
Hi @maitry63, thanks for confirming the reproduction!

I actually already worked on a fix for this — opened PR #23221 which
normalizes class_weight keys to integers (so string/float keys like "0" or
0.0 work correctly) and raises a clear ValueError for genuinely invalid
keys instead of silently defaulting to 1.0. I also added unit tests
covering these cases in data_adapter_utils_test.py.

All CI checks are passing. Would really appreciate a review when you have a
chance: https://github.com/keras-team/keras/pull/23221
